src/vfvm_system.jl

```diff
@@ -103,10 +103,14 @@ mutable struct System{Tv, Tc, Ti, Tm, TSpecMat <: AbstractMatrix} <: AbstractSys
     generic_matrix::SparseMatrixCSC
 
     """
-    Sparse matrix colors for generic operator handling
+    Sparse matrix colors etc. for generic operator handling
     """
-    generic_matrix_colors::Vector
+    generic_matrix_prep::DifferentiationInterface.SparseJacobianPrep
```

Comment:
This is type-unstable, and in general preparation results are internal implementation details of DI, which can be renamed without warning. It's better to leave this as:
```suggestion
    generic_matrix_prep::P
```
and parametrize `P`

Reply:
Thanks for the advise! There is some more refactoring going on, making System type stable will be topic of a later PR.

Reply:
Apart from type-stability, this one is also problematic because you are using an internal type of DifferentiationInterface. `DI.SparseJacobianPrep` is not a public symbol, it may be renamed or removed without warning and that will break your code. Thus it is important not to rely on it (or promise not to complain if your code does break because of that one day ;) )